index.js

```diff
@@ -145,16 +145,34 @@ function $asInteger (i) {
 }
 
 function $asNumber (i) {
+  if (i === 0) return '0'
+  if (!i || i === true) {
+    throw new Error('Cannot coerce to number')
+  }
+  if (i.constructor === Number) {
+    if (!isFinite(i)) {
+      throw new Error('Cannot coerce to number')
+    } else {
+      return '' + i
+    }
+  }
+  if (typeof i === 'object') {
+    throw new Error('Cannot coerce to number')
+  }
   var num = Number(i)
-  if (isNaN(num)) {
-    return 'null'
+  if (!isFinite(num)) {
+    throw new Error('Cannot coerce to number')
   } else {
     return '' + num
   }
 }
 
 function $asBoolean (bool) {
-  return bool && 'true' || 'false' // eslint-disable-line
+  if (bool === true) return 'true'
+  if (bool === false) return 'false'
+  if (bool === 'true') return 'true'
+  if (bool === 'false') return 'false'
```

Comment:
I'm __very__ -1 on rendering a string of `'true'` as boolean `true`.

Reply:
Truly I replicate the *same* approach used for number: 
- "55" => 55
- "true" => true

Let me know if I'll remove those `if`s

Reply:
I'm not 100% sure we are doing the right thing.

@delvedor would you prefer this to be strict about the type?